sound/soc/sof/loader.c

```diff
@@ -872,7 +872,8 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	}
 
 	/* fw boot is complete. Update the active cores mask */
-	sdev->enabled_cores_mask = init_core_mask;
+	/* Primary core is active no matter init_core_mask is set or not */
+	sdev->enabled_cores_mask = init_core_mask | BIT(SOF_DSP_PRIMARY_CORE);
```

Comment:
Sorry, the commit message may be misleading. What I tried to say it that chip->init_core_mask is not set in all platforms.  In other words, chip->init_core_mask could be 0. That's why I tried to add the  SOF_DSP_PRIMARY_CORE bit to the  enabled_cores_mask. I agree that we don't need to`` | BIT(SOF_DSP_PRIMARY_CORE)`` in ``cl_dsp_init()`` since all platforms which set ``.run = hda_dsp_cl_boot_firmware,`` or ``.run = hda_dsp_cl_boot_firmware_iccmax,`` also set ``.init_core_mask``, too. But to let ``enabled_cores_mask`` match the real core power status, I think we should set  ``sdev->enabled_cores_mask = init_core_mask | BIT(SOF_DSP_PRIMARY_CORE)`` here.

Reply:
I think we shuldnt assume this but instead check if init_core_mask returned after FW boot is non-zero

Reply:
we will lose when enabled_core_mask is set before this. Actually ICL ICCMAX, we need to enable/stall core 3 before this. hda_dsp_cl_boot_firmware return primary core id. How about changing this to

```
-       sdev->enabled_cores_mask = init_core_mask | BIT(SOF_DSP_PRIMARY_CORE);
+       sdev->enabled_cores_mask |= init_core_mask;
```